src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -806,3 +809,19 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
         }
     # endregion
 
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain the 10 status complete file but remove older .complete.status files """
+        list_of_files = glob.glob(status_folder + '/' + '*.complete.status')
+        files_to_removed = []
+
+        if len(list_of_files) <= Constants.MAX_COMPLETE_STATUS_FILES_COUNT:
+            return
+
+        while len(list_of_files) > Constants.MAX_COMPLETE_STATUS_FILES_COUNT:
+            oldest_file = min(list_of_files, key=lambda x: (os.path.getmtime(x), int(re.search(r'(\d+)\.complete.status', x).group(1)), x))
+            file_base_name = os.path.basename(oldest_file)
+            files_to_removed.append(oldest_file)
+            self.env_layer.file_system.delete_files_from_dir(status_folder, [file_base_name])
+            list_of_files.remove(oldest_file)
+
+        self.composite_logger.log_debug("Cleaned up older complete status files: {0}".format(files_to_removed))
```

Comment:
![image](https://github.com/Azure/LinuxPatchExtension/assets/56841542/6a556173-89ec-4f40-8f43-6fb9a6af1c16)

Is this validated?

Reply:
validated check screenshot